### DISCUSSION TITLE
Is that possible to input the notes name when use a template to create a new note?

### DISCUSSION THREAD

**Discussion by francisHugo** [mProjectsCode/obsidian-meta-bind-plugin#250] [Q&A]
I use the template to create new note, but I found the new note name is a set string in `fileName: a set string`. and can only by renamed after the note is been created.

I would like to be able to input the note name when create it by input the note  name.

is it possible to input the note name instead of the `a set string`?

**Comment by mProjectsCode** [comment]
Not directly with buttons, but you can rename the note in your templater template.

**Reply by samrauh** [reply]
I tried this but sadly it is not working. When I try to rename the note in the template using `<% tp.file.rename("Test Name") %>` all the other lines from the template are missing. It also fails to automatically set properties using templater code

**Comment by 3hree6ixty5ive** [comment]
I use the method mProjectCode mentioned and rename it inside the template using this code:

<%*
let name = await tp.system.prompt("File Name:", null, true);
await tp.file.rename(`${name}`);
await app.workspace.getLeaf(true).openFile(tp.file.find_tfile(`${name}`));
-%>

You might not need the last line, as that is just for my use case of opening it in a new pane.
